$MirOS: ports/mailnews/pine/patches/patch-imap_src_osdep_unix_ssl_unix_c,v 1.2 2015/07/18 23:42:13 tg Exp $
--- imap/src/osdep/unix/ssl_unix.c.orig	Tue Apr 27 19:54:29 2004
+++ imap/src/osdep/unix/ssl_unix.c	Wed May  4 23:28:08 2005
@@ -89,7 +89,8 @@ void ssl_onceonlyinit (void)
     struct stat sbuf;
 				/* if system doesn't have /dev/urandom */
     if (stat ("/dev/urandom",&sbuf)) {
-      if ((fd = open (tmpnam (tmp),O_WRONLY|O_CREAT,0600)) < 0)
+      strlcpy(tmp, "/tmp/cclient.XXXXXXXXXX", sizeof(tmp));
+      if ((fd = mkstemp(tmp)) < 0)
 	i = (unsigned long) tmp;
       else {
 	unlink (tmp);		/* don't need the file */
